File: app/src/test/java/fr/nspu/dev/recordvoicelearning/controller/converter/QuestionToAnswerEnumConverterTest.java
package fr.nspu.dev.recordvoicelearning.controller.converter;

import org.junit.Test;

import fr.nspu.dev.recordvoicelearning.utils.QuestionToAnswerEnum;

import static org.junit.Assert.*;

/**
 * Created by nspu on 18-02-23.
 */
public class QuestionToAnswerEnumConverterTest {
    @Test
    public void toQuestionToAnswerEnum() throws Exception {
        QuestionToAnswerEnum questionToAnswerEnumTrue = QuestionToAnswerEnumConverter.Companion.toQuestionToAnswerEnum(true);
        QuestionToAnswerEnum questionToAnswerEnumFalse = QuestionToAnswerEnumConverter.Companion.toQuestionToAnswerEnum(false);

        assertEquals(questionToAnswerEnumTrue, QuestionToAnswerEnum.QUESTION_TO_ANSWER);
        assertEquals(questionToAnswerEnumFalse, QuestionToAnswerEnum.ANSWER_TO_QUESTION);
    }

    @Test
    public void toBoolean() throws Exception {
        boolean bTrue = QuestionToAnswerEnumConverter.Companion.toBoolean(QuestionToAnswerEnum.QUESTION_TO_ANSWER);
        boolean bFalse = QuestionToAnswerEnumConverter.Companion.toBoolean(QuestionToAnswerEnum.ANSWER_TO_QUESTION);
        boolean bDefault = QuestionToAnswerEnumConverter.Companion.toBoolean(null);

        assertEquals(bTrue, true);
        assertEquals(bFalse, false);
        assertEquals(bDefault, true);
    }

}

File: app/src/test/java/fr/nspu/dev/recordvoicelearning/controller/converter/QuestionToAnswerEnumConverterTest.kt
package fr.nspu.dev.recordvoicelearning.controller.converter

import fr.nspu.dev.recordvoicelearning.utils.QuestionToAnswerEnum
import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Created by nspu on 18-02-23.
 */
class QuestionToAnswerEnumConverterTest {
    val questionToAnswerEnumConverter = QuestionToAnswerEnumConverter()

    @Test
    @Throws(Exception::class)
    fun toQuestionToAnswerEnum() {
        val questionToAnswerEnumTrue = questionToAnswerEnumConverter.toQuestionToAnswerEnum(true)
        val questionToAnswerEnumFalse = questionToAnswerEnumConverter.toQuestionToAnswerEnum(false)
        val questionToAnswerEnumFalseDefault = questionToAnswerEnumConverter.toQuestionToAnswerEnum(null)

        assertEquals(questionToAnswerEnumTrue, QuestionToAnswerEnum.QUESTION_TO_ANSWER)
        assertEquals(questionToAnswerEnumFalse, QuestionToAnswerEnum.ANSWER_TO_QUESTION)
        assertEquals(questionToAnswerEnumFalseDefault, QuestionToAnswerEnum.QUESTION_TO_ANSWER)
    }

    @Test
    @Throws(Exception::class)
    fun toBoolean() {
        val bTrue = questionToAnswerEnumConverter.toBoolean(QuestionToAnswerEnum.QUESTION_TO_ANSWER)
        val bFalse = questionToAnswerEnumConverter.toBoolean(QuestionToAnswerEnum.ANSWER_TO_QUESTION)

        assertEquals(bTrue, true)
        assertEquals(bFalse, false)
    }

}